sqlmesh/core/model/definition.py

```diff
@@ -524,14 +525,10 @@ def convert_to_time_column(
     def update_schema(
         self,
         schema: MappingSchema,
-        default_schema: t.Optional[str] = None,
-        default_catalog: t.Optional[str] = None,
     ) -> None:
         """Updates the schema for this model's dependencies based on the given mapping schema."""
         for dep in self.depends_on:
-            table = d.set_default_schema_and_catalog(
-                dep, default_schema=default_schema, default_catalog=default_catalog
-            )
+            table = exp.to_table(dep, dialect=self.dialect)
```

Comment:
this shouldn't call self.dialect because dep is already normalized to sqlglot dialect.

Reply:
line 541 also needs to change, it shouldn't call sql(dialect=self.dialect), if this model has back ticks it should break